packages/server/core/src/core/server.ts

```diff
@@ -0,0 +1,171 @@
+  config: ServerOptions;
+  serverConfigFile?: string;
+  routes?: ServerRoute[];
+  plugins?: ServerPlugin[];
+  internalPlugins?: InternalPlugins;
+  appContext: {
+    appDirectory?: string;
+    sharedDirectory?: string;
+    apiDirectory?: string;
+    lambdaDirectory?: string;
+  };
+  runMode?: 'apiOnly' | 'ssrOnly' | 'webOnly';
+};
+
+export type SSRServerContext = BaseSSRServerContext & {
+  staticGenerate?: boolean;
+};
+
+export type ServerRender = (
+  ssrContext: SSRServerContext,
+) => Promise<string | Readable | ReadableStream>;
+
+export type RenderResult = {
+  // eslint-disable-next-line node/prefer-global/buffer
+  content: string | Buffer;
+  contentType: string;
+  contentStream?: Readable;
+  statusCode?: number;
+  redirect?: boolean;
+};
+
+export type ConfWithBFF = {
+  bff?: {
+    prefix: string;
+  } & ServerOptions['bff'];
+};
+
+export type Then<T> = T extends PromiseLike<infer U> ? U : T;
+
+export type { Metrics, Logger, NextFunction };
+
+export type HookNames = 'afterMatch' | 'afterRender';
+
+export interface ModernServerInterface {
+  pwd: string;
+
+  distDir: string;
+
+  onInit: (runner: ServerHookRunner, app: Server) => Promise<void>;
+
+  getRequestHandler: () => (
+    req: IncomingMessage,
+    res: ServerResponse,
+    next?: () => void,
+  ) => void;
+
+  createHTTPServer: (
+    handler: (
+      req: IncomingMessage,
+      res: ServerResponse,
+      next?: () => void,
+    ) => void,
+  ) => Promise<Server>;
+
+  render: (
+    req: IncomingMessage,
+    res: ServerResponse,
+    url?: string,
+  ) => Promise<string | Readable | null>;
+
+  close?: () => Promise<void>;
+}
+
+export type ServerConstructor = (
+  options: ServerBaseOptions,
+) => ModernServerInterface;
+
+export type ModernServerHandler = (
+  context: ModernServerContext,
+  next: NextFunction,
+) => Promise<void> | void;
+
+export type Middleware<Env extends HonoEnv = any> = MiddlewareHandler<Env>;
+
+export type RequestHandler = (
+  request: Request,
+  ...args: any[]
+) => Response | Promise<Response>;
+
+type ServerLoaderBundle = {
+  routes: NestedRoute[];
+  handleRequest: (options: {
+    request: Request;
+    serverRoutes: ServerRoute[];
+    context: any;
+    routes: NestedRoute[];
+  }) => Promise<any>;
+};
+
+type ServerRenderBundle = {
+  serverRender: () => any;
+};
+
+export type ServerManifest = {
+  loaderBundles?: Record<string, ServerLoaderBundle>;
+  renderBundles?: Record<string, ServerRenderBundle>;
+  loadableStats?: Record<string, any>;
+  routeManifest?: Record<string, any>;
+};
+
+type ServerVariables = {
+  logger: Logger;
+  reporter?: Reporter;
+  serverManifest?: ServerManifest;
+  /**
+   * Communicating with custom server hook & modern ssrContext.
+   *
+   * Produce by custom server.
+   * Custom by ssrRuntime.
+   */
+  locals?: Record<string, any>;
+  metrics?: Metrics;
+  templates?: Record<string, string>;
+};
+
+export type ServerEnv = {
+  Variables: ServerVariables;
+};
+
+export type {
```

Comment:
Because we want to reduce direct references to hono, and different runtimes may have different types, such as ServerNodeContext or ServerWorkerContext

Reply:
done, @GiveMe-A-Name has solve the problem